packages/block-library/src/query-loop/style.scss

```
@@ -0,0 +1,17 @@
+.wp-block-query-loop {
+	max-width: 100%;
+	display: flex;
+	flex-wrap: wrap;
+	flex-direction: column;
+
+	&.is-grid {
+		flex-direction: row;
+	}
+	@include break-small {
+		@for $i from 2 through 6 {
+			&.is-grid.columns-#{ $i } > div {
+				width: calc((100% / #{$i}) - 1em);
+			}
+		}
+	}
+}
```

Switching this to use a CSS-grid instead of CS-Flexbox would be a lot more efficient... Gaps get auto-added and everything just works :)
```suggestion
.wp-block-query-loop {
	max-width: 100%;
	display: grid;
	grid-gap: 1em;

	&.is-grid {
		flex-direction: row;
	}
	@include break-small {
		@for $i from 2 through 6 {
			&.is-grid.columns-#{ $i } {
				grid-template-columns: repeat(#{ $i }, 1fr);
			}
		}
	}
}
```

---

With `grid` I'll still need ` flex-direction` ? (line 7)

---

Nope! Good catch, I removed it :+1:

---

That's perfect! Thanks!

---

While I think Ari is generally right here that Grid is very efficient, especially when you choose to have an explicit column slider, I would suggest (as also indicated in my comment in https://github.com/WordPress/gutenberg/pull/27067#issuecomment-730183286) that we probably shouldn't have a column slider at all. 

Specifically that opens up questions about responsiveness, and how a theme can help influence the grid gap and general layout. Grid columns also have a tendency to be slightly fluid in ways that aren't super predictable.

So while the code is good (I'll repeat that this is an impressive PR!) — it just seems like this kind of column handling needs to be done in a more generic way, so we avoid having to create unique CSS on a per-block basis.